### ISSUE TITLE
Rise/Set date difference by one day?

### ISSUE THREAD

**Issue by vincentneo** [onekiloparsec/SwiftAA#97]
Sorry to bother you again, but here's something I don't quite get. If I were to use the Paris coordinates from the example you gave, on its local time, rise, transit and set dates are of the same day.

However, if I were to try at Singapore or San Francisco coordinates as some examples, on their local time, either the rise or set date is off by one day. Is that supposed to be normal?

I opened this as a separate issue as it's not quite related with the previous one, I feel.

Here's the code I've tried:
``` Swift
    let jd = JulianDay(year: 2020, month: 09, day: 08)
    let paris = GeographicCoordinates(positivelyWestwardLongitude: Degree(.minus, 2, 21, 07), latitude: Degree(.plus, 48, 51, 24))
    let times = Earth(julianDay: jd).twilights(forSunAltitude: 0, coordinates: paris)
    
    let formatter = DateFormatter()
    formatter.dateFormat = "yyyy-MM-dd'T'HH:mm:ssZ"
    formatter.timeZone = TimeZone(abbreviation: "UTC+2")
    print(formatter.string(from: times.riseTime!.date))         // 2020-09-08T07:21:41+0200
    print(formatter.string(from: times.transitTime!.date))      // 2020-09-08T13:46:57+0200
    print(formatter.string(from: times.setTime!.date))          // 2020-09-08T20:11:16+0200

    formatter.timeZone = TimeZone(abbreviation: "UTC+8")
    let singapore = GeographicCoordinates(CLLocation(latitude: 1.290, longitude: 103.852))
    let timesSG = Earth(julianDay: jd).twilights(forSunAltitude: 0, coordinates: singapore)
    print(formatter.string(from: timesSG.riseTime!.date))       // 2020-09-09T07:00:19+0800
    print(formatter.string(from: timesSG.transitTime!.date))    // 2020-09-08T13:01:03+0800
    print(formatter.string(from: timesSG.setTime!.date))        // 2020-09-08T19:01:27+0800
    
    formatter.timeZone = TimeZone(abbreviation: "UTC-7")
    let sf = GeographicCoordinates(CLLocation(latitude: 37.7740, longitude: -122.4313))
    let timesSF = Earth(julianDay: jd).twilights(forSunAltitude: 0, coordinates: sf)
    print(formatter.string(from: timesSF.riseTime!.date))       // 2020-09-08T06:49:19-0700
    print(formatter.string(from: timesSF.transitTime!.date))    // 2020-09-08T13:05:58-0700
    print(formatter.string(from: timesSF.setTime!.date))        // 2020-09-07T19:23:31-0700
```

**Comment by onekiloparsec** [comment]
No bothering at all, things get improved when serious users ask questions. I'll have a look.

**Comment by vincentneo** [comment]
I also noticed that post midnight of my local time, rise, set, transit values (regardless +/- 1 day) will be 1 day ahead for all three values. Seems to apply to timezones that is one day behind my local time (My timezone is sg's UTC+8 is 10 sep 12:28am, I checked against london's UTC+1, which is at 9 sep 5.28pm)

Using `JulianDay(Date())` for the above scenario, where I thought `Date()` was to represent a specific point of time regardless of timezones, so it kinda shocked me.

Do advise if my understanding of Date() is wrong. Thank you!
